view/tabs/memdump.js

```diff
@@ -133,3 +152,69 @@ $("#memdump").on("click", ".dumpline", function() {
         toggleBreakpoint(brkaddr);
     }
 });
+
+
+$("#dumpcontent").on("focusin focusout keyup keydown", ".membytes div[contenteditable]", function(evt) {
+    var $this = $(this);
+    var value = $this.text();
+    switch(evt.type) {
+        case 'keyup':
+        case 'keydown':
+            if(evt.keyCode >= 8 && evt.keyCode <= 9) return; // Backspace, Tab
+            if(evt.keyCode == 13) { // ENTER
+                evt.preventDefault();
+                $this.blur();
+                return;
+            }
+
+            if(!/^[a-fA-F0-9]+$/.test(value)) {
+                // TODO: limit input to valid HEX values
+            }
+
+            if(value.length > 2) {
+                evt.preventDefault();
+                const t = $this.text().substr(2,1);
+                $this.text($this.text().substr(0,2));
+                $next = $this.next();
+                $next.text(t).trigger('focus');
+            }
+
+            break;
+        case 'focusin':
+            var sel, range;
+            range = document.createRange();
+            if(value.length == 2) {
+                range.selectNodeContents(evt.currentTarget);
+            } else {
+                range.setStart(evt.currentTarget, 1);
+                // range.setEnd(evt.currentTarget, 0);
+            }
+            sel = window.getSelection();
+            sel.removeAllRanges();
+            sel.addRange(range);
+            break;
+        case 'focusout':
+            var addr = $this.data('addr');
+            var previous = hex(parseInt($this.data('byte')), true, 2).toLowerCase();
+            var current = $this.text().toLowerCase();
+            if(current != previous) {
+                var new_value = parseInt(current, 16);
+                zealcom.mem_write(addr, new_value);
+
+                var c = String.fromCharCode(new_value);
+                // ascii = ascii.map((c) => c == ' ' ? '&nbsp;' : c);
+                if(c == ' ') c = '&nbsp;';
+
+                $this.closest('.memline').find(`.asciichars [data-addr=${addr}]`).html(c);
+            }
+            break;
+    }
+});
+
+// $(() => {
```

Comment:
Testing leftover? :smile:

Reply:
yes - I left this in intentionally, to make future edits to the memdump easier ... but I can remove it if you'd like.